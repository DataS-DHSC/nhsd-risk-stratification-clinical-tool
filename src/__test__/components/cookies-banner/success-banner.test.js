import { createEvent, fireEvent, render, screen } from '@testing-library/react';
import SuccessBanner from '../../../components/cookies-banner/success-banner';
import CookiesContextProvider from '../../../context/cookies-context';

const setup = (props) =>
  render(
    <CookiesContextProvider getCookie={() => undefined}>
      <SuccessBanner {...props} />
    </CookiesContextProvider>
  );

describe('Cookies - SuccessBanner', () => {
  it('should match snapshot', () => {
    const props = {
      hideSuccessBanner: jest.fn(),
    };
    const container = setup(props);
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should call hideSuccessBanner once the hide button is pressed', () => {
    const props = {
      hideSuccessBanner: jest.fn(),
    };
    setup(props);
    const element = screen.getByText(/Hide this message/i);
    const clickEvent = createEvent.click(element);
    fireEvent(element, clickEvent);
    expect(clickEvent.defaultPrevented).toBe(true);
    expect(props.hideSuccessBanner).toHaveBeenCalled();
  });
});
