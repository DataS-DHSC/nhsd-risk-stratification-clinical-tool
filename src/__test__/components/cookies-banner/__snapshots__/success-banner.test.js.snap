// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookies - SuccessBanner should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-success-banner"
    id="nhsuk-success-banner"
  >
    <div
      class="nhsuk-width-container"
    >
      <p
        class="nhsuk-success-cookie-banner__message"
        id="nhsuk-success-banner__message"
      >
        You can change your cookie settings at any time using our 
        <a
          href="/page/cookies/policy"
        >
          cookies page
        </a>
        .
      </p>
      <p
        class="nhsuk-success-cookie-banner__message"
        id="nhsuk-success-banner-hide__message"
      >
        <a
          href=""
        >
          Hide this message
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
